### ISSUE TITLE
[Bug]: 因vllm的版本不同，启动的qwen2.5服务，对于相同的输入；0.6.1.post2 sse输出是正确的，但 0.6.3.post1是错误的？

### ISSUE THREAD

**Issue by mawenju203** [vllm-project/vllm#10280]
### Your current environment

<details>
<summary>The output of `python collect_env.py`</summary>

```text

环境：
Vllm==0.6.1.post2
torch==2.4.0
Transformers==4.45.1
Accelerate==0.30.1
Tiktoken==0.7.0

startup:
python -m vllm.entrypoints.openai.api_server  --model ./Qwen2.5-7B-Instruct  --port 8009 --enable-auto-tool-choice --tool-call-parser hermes --max-seq-len-to-capture 2048 --uvicorn-log-level warning

output:

版本： 0.6.3.post1
<img width="783" alt="企业微信截图_1731321799635" src="https://github.com/user-attachments/assets/f34b47e0-dbdd-4181-8df3-6ccb8a4a91df">

版本：0.6.1.post2
<img width="692" alt="企业微信截图_17313218297996" src="https://github.com/user-attachments/assets/923bd793-8e7c-4359-bb36-5fbfd216ba1c">


```

</details>

### 🐛 Describe the bug

环境：
Vllm==0.6.1.post2
torch==2.4.0
Transformers==4.45.1
Accelerate==0.30.1
Tiktoken==0.7.0

startup:
python -m vllm.entrypoints.openai.api_server  --model ./Qwen2.5-7B-Instruct  --port 8009 --enable-auto-tool-choice --tool-call-parser hermes --max-seq-len-to-capture 2048 --uvicorn-log-level warning

output:

版本： 0.6.3.post1
<img width="783" alt="企业微信截图_1731321799635" src="https://github.com/user-attachments/assets/f34b47e0-dbdd-4181-8df3-6ccb8a4a91df">

版本：0.6.1.post2
<img width="692" alt="企业微信截图_17313218297996" src="https://github.com/user-attachments/assets/923bd793-8e7c-4359-bb36-5fbfd216ba1c">

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by jeejeelee** [comment]
For issues like this, I usually suggest first ruling out whether it's caused by a cudagraph bug. You can try adding `--enforce-eager` to verify this

**Comment by uzhao** [comment]
> For issues like this, I usually suggest first ruling out whether it's caused by a cudagraph bug. You can try adding `--enforce-eager` to verify this

I got this issue for Qwen2.5-Coder-0.5B-Instruct-GGUF, while AWQ return normally.

**Comment by jeejeelee** [comment]
> > For issues like this, I usually suggest first ruling out whether it's caused by a cudagraph bug. You can try adding `--enforce-eager` to verify this
> 
> I got this issue for Qwen2.5-Coder-0.5B-Instruct-GGUF with enforce-eager, while AWQ return normally.

Could you please provibe your running script? I am happy to help reproduce your issue

**Comment by uzhao** [comment]
> > > For issues like this, I usually suggest first ruling out whether it's caused by a cudagraph bug. You can try adding `--enforce-eager` to verify this
> > 
> > 
> > I got this issue for Qwen2.5-Coder-0.5B-Instruct-GGUF with enforce-eager, while AWQ return normally.
> 
> Could you please provibe your running script? I am happy to help reproduce your issue

I ran with this command
docker run --runtime nvidia --gpus all \
  -v /home/papillon/huggingface:/root/.cache/huggingface \
  -p 8000:8000 \
  --ipc=host \
  vllm/vllm-openai:latest \
  --model /root/.cache/huggingface/test \
  --served-model-name test

Models are directly from huggingface. 
huggingface-cli download Qwen/Qwen2.5-Coder-0.5B-Instruct-AWQ --local-dir ./test
huggingface-cli download Qwen/Qwen2.5-Coder-0.5B-Instruct-GGUF qwen2.5-coder-0.5b-instruct-q4_k_m.gguf --local-dir ./test

**Comment by jeejeelee** [comment]
> > > > For issues like this, I usually suggest first ruling out whether it's caused by a cudagraph bug. You can try adding `--enforce-eager` to verify this
> > > 
> > > 
> > > I got this issue for Qwen2.5-Coder-0.5B-Instruct-GGUF with enforce-eager, while AWQ return normally.
> > 
> > 
> > Could you please provibe your running script? I am happy to help reproduce your issue
> 
> I ran with this command docker run --runtime nvidia --gpus all -v /home/papillon/huggingface:/root/.cache/huggingface -p 8000:8000 --ipc=host vllm/vllm-openai:latest --model /root/.cache/huggingface/test --served-model-name test
> 
> Models are directly from huggingface. huggingface-cli download Qwen/Qwen2.5-Coder-0.5B-Instruct-AWQ --local-dir ./test huggingface-cli download Qwen/Qwen2.5-Coder-0.5B-Instruct-GGUF qwen2.5-coder-0.5b-instruct-q4_k_m.gguf --local-dir ./test
> 
> GGUF returned !*512

I can basically reproduce your generated results, but I also can generate the reasonable result by using `qwen2.5-7b-instruct-q2_k.gguf`.  @Isotr0py  could you please look at this issue?

**Comment by Isotr0py** [comment]
Seems that there is something wrong in gguf's MMQ kernel, the output of MMQ kernel calling contains NaN, and I can generate reasonable after disabling MMQ and forced to use dequantize and mulmat.

**Comment by Mayflyyh** [comment]
@Isotr0py Hi, I’ve noticed this bug as well. Are you currently working on resolving it? If you need any assistance, I’d be happy to help.

**Comment by Isotr0py** [comment]
@Mayflyyh Yes, I'm still keeping an eye on this issue. But I need more time to figure out which `ggml_type` is suffering from this, because I noticed some quantization like `q6_k_m` didn't encounter this issue.

Some precision like k-quant and imatrix is combined with multiple `ggml_type`, which is causing the debug difficult. (The `NaN` is only occurring in middle decoder layer) 😅
